lock-api/src/main/java/com/palantir/lock/client/TransactionStarter.java

```diff
@@ -70,12 +71,17 @@ static TransactionStarter create(LockLeaseService lockLeaseService, LockWatchEve
     }
 
     StartIdentifiedAtlasDbTransactionResponse startIdentifiedAtlasDbTransaction() {
-        try {
-            return autobatcher.apply(null).get();
-        } catch (ExecutionException e) {
-            throw Throwables.throwUncheckedException(e.getCause());
-        } catch (Throwable t) {
-            throw Throwables.throwUncheckedException(t);
+        return AtlasFutures.getUnchecked(autobatcher.apply(null));
```

Comment:
Can someone explain why do we have this bizarre interface here with passing nulls?

Reply:
Hmm, I guess we sometimes need to pass an actual argument here.

Reply:
I think there's an argument for rewriting this to pass actual number of requested transactions to start? So that internal ski product's disruptors would actually become extremely tiny! Then the batch processor could just count number of transactions requested across the batch and send that to timelock?

1. we would save on a bunch of allocations for futures and lists for args here
2. We would get rid of the weird "null" passing of arguments?

@jeremyk-91 interested in your thoughts

Reply:
Would you mean that, instead of passing `null`, you pass in say `count`, and then the autobatcher batches several `count`s together into a larger batch and fans those out? If so, that would definitely reduce the size of the autobatcher (or at least, the ring buffer underneath).

Reply:
Rewriting to add numbers makes sense - as noted `null` is just because we need to pass _something_. This isn't really a thing raw AtlasDB users would think about as much, but I see how it's useful in ski product (though let's do that in a separate self-contained piece please)